### PR TITLE
develop fixups, and .gitignore fixups

### PR DESCRIPTION
**Pull request by nslater**

### CODE REVIEW

File: DEVELOP.rst [hunk 1]
```diff
@@ -24,17 +24,33 @@ Then, run::
...
-If you want to build the documentation of CrateDB you will need to install
-the ``sphinx-csv-filter`` package upfront::
+Now, install the ``sphinx-csv-filter`` package::
 
     $ pip install sphinx-csv-filter
 
+To work around an issue with conflicting Python paths, copy the ``sphinx``
+module to the main source directory::
+
+    $ cp -R env/lib/python3.4/site-packages/crate/sphinx src/crate/sphinx
```
Comment (chaudum) [inline DEVELOP.rst]: this does not make sense to me. what does it solve? if there is a problem with paths, it should be fixed.

Comment (nslater) [inline DEVELOP.rst]: the problem is, as far as I can tell, that under `src`, you have `crate.theme`. but then the theme wants to import `crate.sphinx` which lives under the `env` directory structure. and I think this confuses Python? having two `crate` dirs on the PYTHONPATH, each one contributing a different module. so my ugly bodge was just to copy over the module

Comment (nslater) [inline DEVELOP.rst]: @chaudum would you be happy with this if I annotated it to make it clear that that it's a hackish workaround?

Comment (chaudum) [inline DEVELOP.rst]: the solution would be that both of the packages are namespace packages. funnily they are already :| guess I need to check what the problem is then.

**Comment by nslater** [comment]
@chaudum okay, all of this works! we're still linking a python module directory. but this is more elegant than the last approach, and everything works now

I also added a new commit to sort the .gitignore file. I do not intend to merge this when rebasing, as it is an unrelated change to the develop file

**Comment by nslater** [comment]
@chaudum added a new section to the DEVELOP file detailing how to rebuild the docs

File: DEVELOP.rst [hunk 2]
```diff
@@ -88,7 +102,39 @@ If you want to check the PyPI description before uploading, run this::
...
+environment for that version, allowing RTD to pick up the newest theme.
+
+Once done, go to the Builds_ page, the select the version you want to rebuild
+from the drop-down menu, and select *Build Version*.
+
+Once the docs have been rebuilt (you can refresh this page to get the current
+status) you can switch back to the Versions_ page and click on the version to
+view the docs as they appear on RTD.
+
+The Crate.io website sits in front of Fastly, which aggressively caches all
```
Comment (chaudum) [inline DEVELOP.rst]: ~~in front~~ behind?

**Comment by nslater** [comment]
thank you! merging!

**Pull request merged**